client/pages/monthly-expense-list.jsx

```diff
@@ -18,7 +29,102 @@ class MonthlyExpenseList extends React.Component {
       .catch(err => console.error(err));
   }
 
+  handleClickCloseModal() {
+    this.setState({ modalDisplay: false });
+  }
+
+  handleClick(event) {
+    const list = this.state.categoryList;
+
+    const userId = 1;
+    let date = event.target.id.split(' ');
+    date = date.filter(value => value !== '');
+
+    const categoryName = [];
+    let spentOnCategory = [];
+    for (let i = 0; i < list.length; i++) {
+      categoryName.push(list[i].name);
+      spentOnCategory.push({
+        id: list[i].categoryId,
+        value: 0
+      });
+    }
+
+    let totalExpense = 0;
+
+    fetch(`/api/monthly-expense-graph/${userId}/${date[0]}/${date[1]}`)
+      .then(response => response.json())
+      .then(list => {
+
+        for (let i = 0; i < list.length; i++) {
+          for (let k = 0; k < spentOnCategory.length; k++) {
+            if (list[i].categoryId !== spentOnCategory[k].id) {
+              continue;
+            } else {
+              spentOnCategory[k].value += parseFloat(list[i].amount);
+              break;
+            }
+          }
+        }
+
+        spentOnCategory = spentOnCategory.map(category => category.value);
+
+        for (let j = 0; j < spentOnCategory.length; j++) {
+          totalExpense += spentOnCategory[j];
+        }
```

Comment:
It looks like the client has to do a lot of work to get the data in the right format. Try moving this logic to the server so that the client can just use the data right away.